GitCommands/Git/WorkTree/ListWorkTreeOutputParser.cs

```diff
@@ -0,0 +1,80 @@
+using System.Collections.Generic;
+using System.IO;
+using GitCommands.Git.Commands;
+using GitExtUtils;
+using Microsoft;
+
+namespace GitCommands.Git.WorkTree
+{
+    /// <summary>
+    /// Provides a parser for output of <see cref="GitCommandHelpers.ListWorkTreeCmd"/> command.
+    /// </summary>
+    public class ListWorkTreeOutputParser
+    {
+        /// <summary>
+        /// Parse the output from <see cref="GitCommandHelpers.ListWorkTreeCmd"/> command
+        /// </summary>
+        /// <param name="listWorkTreeCommandOutput">An output of <see cref="GitCommandHelpers.ListWorkTreeCmd"/> command.</param>
+        /// <returns>list with the parsed GitWorkTree.</returns>
+        /// <seealso href="https://git-scm.com/docs/git-worktree"/>
+        public static List<GitWorkTree> Parse(string listWorkTreeCommandOutput)
+        {
+            // <summary>
+            // Here are the 3 types of lines return by the `worktree list --porcelain` that should be handled:
+            //
+            // 1:
+            // worktree /path/to/bare-source
+            // bare
+            //
+            // 2:
+            // /worktree /path/to/linked-worktree
+            // /HEAD abcd1234abcd1234abcd1234abcd1234abcd1234
+            // /branch refs/heads/master
+            //
+            // 3:
+            // worktree /path/to/other-linked-worktree
+            // HEAD 1234abc1234abc1234abc1234abc1234abc1234a
+            // detached.
+            // </summary>
+            List<GitWorkTree> result = new();
+            GitWorkTree? currentWorktree = null;
+            foreach (string line in listWorkTreeCommandOutput.LazySplit('\n'))
+            {
+                if (string.IsNullOrWhiteSpace(line))
+                {
+                    continue;
+                }
+
+                string[] strings = line.Split(' ');
+                switch (strings[0])
```

Comment:
Why parse the output combined instead of line by line?

Reply:
Sorry, I do not fully understand what do you mean. Could you please explain?

Reply:
The loop parses line by line without using the knowledge from which line it is on.